### DISCUSSION TITLE
How to hydrate a store in Next js 13(App Router) ?

### DISCUSSION THREAD

**Discussion by TrySquadDF** [vercel/next.js#55540] [App Router]
If with the use of Page directory everything was clear, then with the use of App directory there are questions about how to hydration Store, I have some Store for example it can be Mobx, and I want to receive already filled Store from the server at what stage it should happen if when using App directory we do not have serverSideProps and in server components we can not use `createContex` and `enableStaticRendering` to generate Store (because of the dependence on react hooks in library `mobx-react-lite`).In this regard, I have questions about how to generate server-side Store(supposed solution is to generate a class on the server side using react `server context`, then merge on the client side with normal context and on the client side sign the Store using **Mobx** `makeAutoObservable`).

P.S If there are examples with other State Managers, you are welcome to share them and I would be happy to review them. The main thing I would like to see is a hydrotation process using next js to organize SEO if, this approach is no longer relevant in the context of Next JS I would like to see an atcual one

- [ ] I would like to see options on how to do this with App Router with code examples.

**Comment by iampava** [comment]
+1

**Comment by bkirillv** [comment]
+1

**Comment by tramck** [comment]
+1
